cmd/goprotoc/goprotoc.go

```diff
@@ -56,9 +56,18 @@ func main() {
 				fail(err.Error())
 			}
 		} else {
+			includeSourceInfo := opts.includeSourceInfo
+			// We have to pass SourceCodeInfo to plugins as they expect this information to generate comments.
+			// This is true for the builtin protoc plugins as well.
+			// We could instead do a separate Parse if we wanted but the logic gets very complicated
```

Comment:
This is definitely better -- doing a separate parse would slow it down too much.

FWIW, the way protoc works is that it always collects source info and then just discards it (by setting that field of the file descriptor to nil) when it needs a source-info-less descriptor (such as saving to a descriptor set without `--include_source_info`).

Reply:
BTW, I think this means that descriptor set output could have source info, even without `--include_source_info` (like if you're generating code in addition to a descriptor set).

So, before saving the descriptor set (around line 99 in the old version of the file), you could set the file descriptor's source info to nil.

Reply:
Good point, will update.

Reply:
It turns out you actually already do this setting to nil! So we're all set here.